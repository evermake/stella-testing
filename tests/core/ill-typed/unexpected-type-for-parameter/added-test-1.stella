language core;

fn function_chooser(flag : Bool) -> (fn(Nat) -> Nat){
    return fn(first : (fn(Nat) -> Nat) ) {
        return fn(second : (fn(Nat) -> Nat) ) {
            return if flag then first else second;
        };
    };
}

fn add_once(a : Nat) -> Nat {
    return succ(a);
}

fn add_twice(a : Nat) -> Nat {
    return succ(add_once(a));
}

fn main(n : Nat) -> Nat {
    return function_chooser(false)(add_once)(add_twice)(n);
}