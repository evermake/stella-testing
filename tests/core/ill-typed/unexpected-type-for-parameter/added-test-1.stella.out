An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_PARAMETER]
expected type
  Nat
but got
  fn (Nat) -> Nat
for the parameter first
in function
  fn (first : fn (Nat) -> Nat)
{
  return fn (second : fn (Nat) -> Nat)
  {
    return if flag then first else second
  }
}

========================
Alternative type errors:
========================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  fn (Nat) -> Nat
but got
  fn (fn (Nat) -> Nat) -> fn (fn (Nat) -> Nat) -> fn (Nat) -> Nat
when typechecking the expression
  fn (first : fn (Nat) -> Nat)
{
  return fn (second : fn (Nat) -> Nat)
  {
    return if flag then first else second
  }
}